src/main/java/org/opensearch/security/http/OnBehalfOfAuthenticator.java

```diff
@@ -43,81 +44,57 @@
 import org.opensearch.security.authtoken.jwt.EncryptionDecryptionUtil;
 import org.opensearch.security.user.AuthCredentials;
 
-public class HTTPOnBehalfOfJwtAuthenticator implements HTTPAuthenticator {
+public class OnBehalfOfAuthenticator implements HTTPAuthenticator {
```

Comment:
Still has many opportunities to reuse the JwtVendor within this class rather than duplicate logic, but might be good fast followups rather than block the critical path